ui/app/components/transform-edit-base.js

```diff
@@ -54,45 +40,33 @@ export default Component.extend(FocusOnInsertMixin, {
     }
     return modelPrefix;
   },
+  persist(method, successCallback) {
+    const model = get(this, 'model');
+    return model[method]().then(() => {
+      successCallback(model);
+    });
+  },
 
-  onEscape(e) {
-    if (e.keyCode !== keys.ESC || this.get('mode') !== 'show') {
+  applyChanges(type, callback = () => {}) {
```

Comment:
By moving the createOrUpdate logic out of actions, we can access it from any component that extends this one, which will ensure our behavior is consistent across transform items